### PR TITLE
Update NCEP RDHPCS regtests to use spack-stack 1.5.0

### LINKED ISSUE #1121: Update RDHPCS regtest modules for spack-stack 1.5.0 and scotch/7.0.4
**Is your feature request related to a problem? Please describe.**
Bring the NCEP RDHPCS regtest modules up-to-date with the current `spack-stack` containing the updated `scotch` install.

**Describe the solution you'd like**
The modules should be updated relative to the current ufs-weather-model modulefiles.

**Describe alternatives you've considered**
The modules could be left using the `hpc-stack` settings, and not make use of the new `spack-stack` RDHPCS installations of the bugfix scotch version, `v7.0.4`.  This would involve maintenance of the code manager install of `scotch`.

**Additional context**
NA.

### PR DESCRIPTION
**Pull request by MatthewMasarik-NOAA**
# Pull Request Summary
Updates the NCEP RDHPCS regtest modules to use `spack-stack/1.5.0`, which includes `scotch/7.0.4`.

## Description

WIP

### Issue(s) addressed
- fixes #1121 

### Check list  

- [ ] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [ ] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [ ] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [ ] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested?
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
* Have the matrix regression tests been run (if yes, please note HPC and compiler)?
* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.)
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):

### CODE REVIEW

File: regtests/bin/matrix_cmake_ncep [hunk 1]
```diff
@@ -62,20 +66,18 @@ EOF
...
+    metispath='/scratch1/NCEPDEV/climate/Matthew.Masarik/waves/opt/spack-stack/1.5.0/parmetis-4.0.3/install'
+    modcmake='cmake/3.23.1'
   elif [ $isorion ]
   then
     batchq='slurm'
     basemodcomp='intel/2022.1.2'
     basemodmpi='impi/2022.1.2'
-    hpcstackpath='/work/noaa/epic-ps/role-epic-ps/hpc-stack/libs/intel-2022.1.2/modulefiles/stack'
-    hpcstackversion='hpc/1.2.0'
+    spackstackpath='/work/noaa/epic-ps/role-epic-ps/hpc-stack/libs/intel-2022.1.2/modulefiles/stack'
```
Comment (JessicaMeixner-NOAA) [inline regtests/bin/matrix_cmake_ncep]: What about updating to this: https://github.com/ufs-community/ufs-weather-model/blob/develop/modulefiles/ufs_orion.intel.lua#L5 ?

File: regtests/bin/matrix_cmake_ncep [hunk 2]
```diff
@@ -62,20 +66,18 @@ EOF
...
+    spackstackversion='hpc/1.2.0'
     modcomp='hpc-intel/2022.1.2'
     modmpi='hpc-impi/2022.1.2'
     scotchpath='/work2/noaa/marine/mmasarik/waves/opt/hpc-stack/scotch-v7.0.3/install'
```
Comment (JessicaMeixner-NOAA) [inline regtests/bin/matrix_cmake_ncep]: Can this line be removed?

File: regtests/bin/matrix_cmake_ncep [hunk 3]
```diff
@@ -139,29 +141,28 @@ EOF
...
-  echo "  module load $modnetcdf"                           >> matrix.head
+  echo "  module load $modnetcdfc"                          >> matrix.head
+  echo "  module load $modnetcdff"                          >> matrix.head
   echo "  module load $modbacio"                            >> matrix.head
   echo "  module load $modg2"                               >> matrix.head
   echo "  module load $modw3emc"                            >> matrix.head
   echo "  module load $modesmf"                             >> matrix.head
 
   echo "  export METIS_PATH=${metispath}"                   >> matrix.head
-  echo "  export SCOTCH_PATH=${scotchpath}"                 >> matrix.head
```
Comment (JessicaMeixner-NOAA) [inline regtests/bin/matrix_cmake_ncep:164]: Should we add a "module load scotch"  here or did I miss that elsewhere?

**Comment by MatthewMasarik-NOAA** [comment]
This is a draft.  These are all currently underway.

**Comment by MatthewMasarik-NOAA** [comment]
Closing this draft.  Once testing completes, a new PR will be opened from an updated branch.

**Pull request closed without merging**